packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1011 @@
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.insetTop:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__top", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalInset",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.insetBottom:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__bottom", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalInset",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.bottom:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__bottom", "")
+                );
+                break;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     *  gets the CSS classes to be programmatically applied to the component
+     */
+    private getPositioningStyles = (): {} => {
+        // determine if we should hide the positioner because we don't have data to position it yet
+        // (avoiding flicker)
+        const shouldHide: boolean =
+            (this.state.disabled &&
+                ((this.props.disabled === undefined || this.props.disabled === false) &&
+                    isNil(this.positionerRect) &&
+                    this.getAnchorElement() !== null)) ||
+            (isNil(this.positionerRect) && !this.state.noOberverMode);
+
+        return {
+            opacity: shouldHide ? 0 : undefined,
+            position: "relative",
+            transformOrigin: `${this.state.xTransformOrigin} ${
+                this.state.yTransformOrigin
+            }`,
+            transform: `translate(
+                ${Math.floor(this.state.xTranslate)}px, 
+                ${Math.floor(this.state.yTranslate)}px
+            )`,
+            top: this.state.top === null ? null : `${this.state.top}px`,
+            right: this.state.right === null ? null : `${this.state.right}px`,
+            bottom: this.state.bottom === null ? null : `${this.state.bottom}px`,
+            left: this.state.left === null ? null : `${this.state.left}px`,
+        };
+    };
+
+    /**
+     *  Checks whether component should be disabled or not
+     */
+    private checkComponentConfig = (): void => {
```

Comment:
this name is not intuitive, i would prefer something like "updateDisabledState"

Reply:
yep